File: 02_wizards/ironpython/pyWizards/test_Wizard.py
import clr
import unittest

clr.AddReference('System')
clr.AddReferenceByName('Wizards')

from Wizards import *
from System import Console

class Test_Wizard(unittest.TestCase):

    def test_asDivineWizardCastNecromancySpellsFromBookNotEffective(self):
        animateDead = Spell("Animate Dead", SpellSchool.Necromancy)
        wizard = Wizard(SpellSchool.Divination)
        wizard.Memorize(animateDead)

        self.assertEqual("Animate Dead", wizard.Cast(animateDead))

    def test_asDivineWizardCastIllusionSpellNotMemorizedFumbles(self):
        invisibility = Spell("Invisiblity", SpellSchool.Illusion)
        wizard = Wizard(SpellSchool.Divination)

        self.assertEqual("fumble", wizard.Cast(invisibility))
    
    def test_asDivineWizardCastDivineSpellFromBookIsEffective(self):
        trueSight = Spell("True Sight", SpellSchool.Divination)
        wizard = Wizard(SpellSchool.Divination)
        wizard.Memorize(trueSight)

        self.assertEqual("Effective True Sight", wizard.Cast(trueSight))

unittest.main(exit = False)
Console.ReadKey()

File: 02_wizards/ruby/test/test_wizard.rb

require 'test/unit'
require 'app/wizard'

class TestWizard < Test::Unit::TestCase

	def test_conjurer_castsconjurationspell_memorized_effective
		conjurer = Conjurer.new
		conjurer.memorize Spells::Conjuration::ANIMAL_SUMMONING
		assert_equal "effective animal summoning", conjurer.cast(Spells::Conjuration::ANIMAL_SUMMONING)
	end

	def test_conjurer_castsdivinationspell_memorized_standard
		conjurer = Conjurer.new
		conjurer.memorize Spells::Divination::TRUE_SIGHT
		assert_equal "true sight", conjurer.cast(Spells::Divination::TRUE_SIGHT)
	end

	def test_necromancer_memorizes_other_spells_fumble
		necromancer = Necromancer.new
		necromancer.memorize Spells::Divination::TRUE_SIGHT
		necromancer.memorize Spells::Conjuration::ANIMAL_SUMMONING
		assert_equal "fumble", necromancer.cast(Spells::Necromancy::RAISE_DEAD)
	end



	def test_conjurer_castunmemorizedspell_fumble
		conjurer = Conjurer.new
		assert_equal "fumble", conjurer.cast(Spells::Conjuration::ANIMAL_SUMMONING)
	end


end
